### ISSUE TITLE
Looking for "multinomial Logistic Regression (cross-entropy loss)" in accord like sklearn

### ISSUE THREAD

**Issue by nahibi** [accord-net/framework#297]
hi,
I used this algorithm in a python program.

http://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html
solver: ‘newton-cg’ or ‘lbfgs’
I am looking for a "multinomial Logistic Regression (cross-entropy loss)" algorithm.

Is there a  similar algorithmin accord.net?
http://accord-framework.net/docs/html/T_Accord_Statistics_Models_Regression_MultinomialLogisticRegression.htm
I found this one but it uses a different solver, right?

**Comment by cesarsouza** [comment]
Hi there,

Thanks for opening the issue! As you correctly noticed, the current Multinomial Logistic Regression implementation in the framework does not use l-bfgs or newton-cg. However, there _are_ implementations for l-bfgs and newton-cg in the framework that could be used to implement such multinomial logistic regression model. They are implemented by the [BroydenFletcherGoldfarbShanno](http://accord-framework.net/docs/html/T_Accord_Math_Optimization_BroydenFletcherGoldfarbShanno.htm) and [ConjugateGradient](http://accord-framework.net/docs/html/T_Accord_Math_Optimization_ConjugateGradient.htm) classes, respectively.

Basically, what would have to be done is to implement the cross-entropy loss and their gradient and pass their function definitions to one of the solvers (i.e. by setting their [Function](http://accord-framework.net/docs/html/P_Accord_Math_Optimization_BaseOptimizationMethod_Function.htm) and [Gradient](http://accord-framework.net/docs/html/P_Accord_Math_Optimization_BaseGradientOptimizationMethod_Gradient.htm) function delegates).

While this shouldn't be that hard to do (there are [formulas for the objective function](http://www.cedar.buffalo.edu/~srihari/CSE574/Chap4/Chap4-Part3.pdf) and its [gradient function](https://www.quora.com/What-is-the-gradient-of-the-log-likelihood-function-in-multinomial-logistic-regression) around), unfortunately it has not been implemented yet. Maybe someone would like to give it a try and submit a pull request?

Regards,
Cesar

**Comment by cesarsouza** [comment]
Fixed in 3.5.0.
